include/tcframe/spec/core/StyleConfig.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <tuple>
+#include <utility>
+#include "CommonConfig.hpp"
+
+using std::move;
+using std::tie;
+
+namespace tcframe {
+
+struct StyleConfig {
+    friend class StyleConfigBuilder;
+
+private:
+    bool generateOutputFiles_;
+
+public:
+    bool needsOutputFiles() const {
+        return generateOutputFiles_;
+    }
+
+    bool operator==(const StyleConfig& o) const {
+        return tie(generateOutputFiles_) == tie(o.generateOutputFiles_);
+    }
+};
+
+class StyleConfigBuilder {
+private:
+    StyleConfig subject_;
+
+public:
+    virtual ~StyleConfigBuilder() {}
+
+    StyleConfigBuilder() {
+        subject_.generateOutputFiles_ = CommonConfig::generateOutput();
+    }
+
+    StyleConfigBuilder& NoOutputFiles() {
```

Comment:
Mmmm... I proposed the `...Files()` suffix to differentiate it between the concept of whole test case files as in the `--output` CLI arg. But, considering that we always just use `output` to refer to `.out` everywhere in the code, and also one possible style is `OutputOnly` and not `OutputFilesOnly`, let's just use `NoOutput()` here :D

You should also replace all occurrences of `generateOutput` etc to `needsOutput` for consistency.